クラッシュダンプの出力ファイル名がおかしい

<!-- 説明に必要なスクリーンショットがあれば貼り付けお願いします。-->
<!-- 画像ファイルをこの欄にドラッグ＆ドロップすれば画像が貼り付けられます -->
クラッシュダンプの出力ファイル名がおかしいです。

https://github.com/sakura-editor/sakura/blob/1ff0b488e5f11a074127447d4c587d7d914afda6/sakura_core/_main/CProcess.cpp#L126-L128

#1735 の指摘で気付いたんですけど `_APP_NAME(_T)` には「sakura(デバッグ版)」のような文字列が入ります。
コードに付けられたコメントと一致しないので、コメントかコードかを修正する必要があります。

「sakura(デバッグ版)」になるのは、`_GSTR_APPNAME_`ですね。
`_APP_NAME(_T)`　は `_T("sakura")` で固定でデバッグ版でも同じです。
ただし、 2018年12月 ad01d9dc 「fixes #709: GSTR_APPNAME の実体をグローバル変数に変える」 でマクロ定義が
sakura_core/config/app_constants.h から sakura_core\_main\WinMain.cpp に移動しており、これ以降`_APP_NAME`マクロがCProcess.cppのコンパイル時には見つからないためにクラッシュダンプがコンパイルエラーになります。
クラッシュダンプは2013年4月の 07cf7fbb の時点ではデフォルトで無効です。
2015年5月 d627eb2a patchunicode#2268 r4023 でリリースビルドでも有効化されていますが
2015年9月 9ca9366e patchunicode#2268(同上)  r4036 でmingwでのヘッダ定義不足のためにifdefが定義されてmingwを除外しようとしています(が少しおかしいようです)
https://github.com/sakura-editor/sakura/blob/1ff0b488e5f11a074127447d4c587d7d914afda6/sakura_core/_main/CProcess.h#L26-L28

とMINIDUMP_TYPEをifdefしていますが、MINIDUMP_TYPEはそもそもenumであって、プリプロセッサマクロ定義ではないのでifdefでは検出できないはずで、つまりVCのみ有効化のつもりが、この時点でVC++でもUSE_MINIDUMPを手動定義しないかぎり、リリースビルドでもデバッグ版でも無効になっているものと考えられます。

ついでにいえば、_Tマクロ切り替えは廃止されているので、_Tになってるのも、修正したほうがよさそうです。
L文字列とANSI文字列の結合はL文字列とみなす、というC++仕様がある(はずな)ので問題になることはなさそうですが、気持ち悪いといえばそうです。

ソースコード上などでUSE_CRASHDUMPを定義すると、テストコードでコンパイルエラーが出ます。
dbghelp.hまたはimagehlp.hのインクルードが必要そうです。
1>(省略...)\main\sakura_core\_main\cprocess.h(72): error C2061: 構文エラー: 識別子 'MINIDUMP_TYPE' (ソース ファイルをコンパイルしています test-csakuraenvironment.cpp)
1>(省略...)\main\sakura_core\_main\cprocess.h(72): error C2061: 構文エラー: 識別子 'MINIDUMP_TYPE' (ソース ファイルをコンパイルしています test-cdlgprofilemgr.cpp)
1>(省略...)\main\sakura_core\_main\cprocess.h(72): error C2061: 構文エラー: 識別子 'MINIDUMP_TYPE' (ソース ファイルをコンパイルしています test-file.cpp)
1>(省略...)\main\sakura_core\_main\cprocess.h(72): error C2061: 構文エラー: 識別子 'MINIDUMP_TYPE' (ソース ファイルをコンパイルしています test-winmain.cpp)

--- a/sakura_core/_main/CProcess.cpp
+++ b/sakura_core/_main/CProcess.cpp
@@ -37,9 +37,6 @@ CProcess::CProcess(
 )
 : m_hInstance( hInstance )
 , m_hWnd( 0 )
-#ifdef USE_CRASHDUMP
-, m_pfnMiniDumpWriteDump(NULL)
-#endif
 {
 	// アプリ名をリソースから読み込む
 	m_strAppName = LS(STR_GSTR_APPNAME);
@@ -89,79 +86,12 @@ bool CProcess::Run()
 {
 	if( InitializeProcess() )
 	{
-#ifdef USE_CRASHDUMP
-		HMODULE hDllDbgHelp = LoadLibraryExedir( L"dbghelp.dll" );
-		m_pfnMiniDumpWriteDump = NULL;
-		if( hDllDbgHelp ){
-			*(FARPROC*)&m_pfnMiniDumpWriteDump = ::GetProcAddress( hDllDbgHelp, "MiniDumpWriteDump" );
-		}
-
-		__try {
-#endif
 			MainLoop() ;
 			OnExitProcess();
-#ifdef USE_CRASHDUMP
-		}
-		__except( WriteDump( GetExceptionInformation() ) ){
-		}
-
-		if( hDllDbgHelp ){
-			::FreeLibrary( hDllDbgHelp );
-			m_pfnMiniDumpWriteDump = NULL;
-		}
-#endif
 		return true;
 	}
 	return false;
 }
-
-#ifdef USE_CRASHDUMP
-/*!
-	@brief クラッシュダンプ
-	
-	@author ryoji
-	@date 2009.01.21
-*/
-int CProcess::WriteDump( PEXCEPTION_POINTERS pExceptPtrs )
-{
-	if( !m_pfnMiniDumpWriteDump )
-		return EXCEPTION_CONTINUE_SEARCH;
-
-	static WCHAR szFile[MAX_PATH];
-	// 出力先はiniと同じ（InitializeProcess()後に確定）
-	// Vista以降では C:\Users\(ユーザー名)\AppData\Local\CrashDumps に出力
-	GetInidirOrExedir( szFile, _APP_NAME_(_T) L".dmp" );
-
-	HANDLE hFile = ::CreateFile(
-		szFile,
-		GENERIC_WRITE,
-		0,
-		NULL,
-		CREATE_ALWAYS,
-		FILE_ATTRIBUTE_NORMAL | FILE_FLAG_WRITE_THROUGH,
-		NULL);
-
-	if( hFile != INVALID_HANDLE_VALUE ){
-		MINIDUMP_EXCEPTION_INFORMATION eInfo;
-		eInfo.ThreadId = GetCurrentThreadId();
-		eInfo.ExceptionPointers = pExceptPtrs;
-		eInfo.ClientPointers = FALSE;
-
-		m_pfnMiniDumpWriteDump(
-			::GetCurrentProcess(),
-			::GetCurrentProcessId(),
-			hFile,
-			MiniDumpNormal,
-			pExceptPtrs ? &eInfo : NULL,
-			NULL,
-			NULL);
-
-		::CloseHandle(hFile);
-	}
-
-	return EXCEPTION_CONTINUE_SEARCH;
-}
-#endif
 
 /*!
 	言語選択後に共有メモリ内の文字列を更新する

--- a/sakura_core/_main/CProcess.h
+++ b/sakura_core/_main/CProcess.h
@@ -25,10 +25,6 @@
 #include "util/design_template.h"
 #include "env/CShareData.h"
 
-#ifdef MINIDUMP_TYPE
-#define USE_CRASHDUMP
-#endif
-
 /*-----------------------------------------------------------------------
 クラスの宣言
 -----------------------------------------------------------------------*/
@@ -52,9 +48,7 @@ protected:
 
 protected:
 	void			SetMainWindow(HWND hwnd){ m_hWnd = hwnd; }
-#ifdef USE_CRASHDUMP
-	int				WriteDump( PEXCEPTION_POINTERS pExceptPtrs );
-#endif
+
 public:
 	HINSTANCE		GetProcessInstance() const{ return m_hInstance; }
 	CShareData&		GetShareData()   { return m_cShareData; }
@@ -67,17 +61,6 @@ public:
 private:
 	HINSTANCE	m_hInstance;
 	HWND		m_hWnd;
-#ifdef USE_CRASHDUMP
-	BOOL (WINAPI *m_pfnMiniDumpWriteDump)(
-		HANDLE hProcess,
-		DWORD ProcessId,
-		HANDLE hFile,
-		MINIDUMP_TYPE DumpType,
-		PMINIDUMP_EXCEPTION_INFORMATION ExceptionParam,
-		PMINIDUMP_USER_STREAM_INFORMATION UserStreamParam,
-		PMINIDUMP_CALLBACK_INFORMATION CallbackParam
-		);
-#endif
 	CShareData		m_cShareData;
 	std::wstring	m_strAppName;
 };

